cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -399,9 +399,14 @@ public void deleteAppVersion(HttpRequest request, HttpResponder responder,
                                @PathParam("namespace-id") final String namespaceId,
                                @PathParam("app-id") final String appId,
                                @PathParam("version-id") final String versionId) throws Exception {
-    ApplicationId id = validateApplicationVersionId(namespaceId, appId, versionId);
-    applicationLifecycleService.removeApplication(id);
-    responder.sendStatus(HttpResponseStatus.OK);
+    if (configuration.getBoolean(Constants.AppFabric.APP_VERSION_DELETION_ENABLED, false)) {
```

Comment:
you can change this to 
```
if (!configuration.getBoolean(Constants.AppFabric.APP_VERSION_DELETION_ENABLED, false)) {
  responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Deletion of specific app version is not allowed.");
  return;
}
...
```
so there will be fewer nested block

Reply:
+1.

Also, by convention, we don't put a default value for anything that is already in cdap-default.xml, since it is always expected to be in the cconf.